packages/react-instantsearch-core/src/core/createInstantSearch.js

```diff
@@ -5,18 +5,14 @@ import version from './version';
 
 /**
  * Creates a specialized root InstantSearch component. It accepts
- * an algolia client and a specification of the root Element.
- * @param {function} defaultAlgoliaClient - a function that builds an Algolia client
+ * a specification of the root Element.
  * @param {object} root - the defininition of the root of an InstantSearch sub tree.
  * @returns {object} an InstantSearch root
  */
-export default function createInstantSearch(defaultAlgoliaClient, root) {
+export default function createInstantSearch(root) {
   return class CreateInstantSearch extends Component {
     static propTypes = {
-      algoliaClient: PropTypes.object,
-      searchClient: PropTypes.object,
-      appId: PropTypes.string,
-      apiKey: PropTypes.string,
+      searchClient: PropTypes.object.isRequired,
```

Comment:
In InstantSearch.js we throw if the `searchClient` does not contain a `search` function:

https://github.com/algolia/instantsearch.js/blob/21e4d81b3e37c03686acaf3a8dc2c73c3c519570/src/lib/InstantSearch.js#L74-L80

We should probably do the same in React InstantSearch via PropTypes.

You can use the Angular InstantSearch [`SearchClient`](https://github.com/algolia/angular-instantsearch/blob/1c0766735a4fcd54854da43f5a83494007b236bf/src/instantsearch/instantsearch.ts#L149-L155) type as reference.